pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go

```diff
@@ -191,6 +191,9 @@ func (c *Controller) updateClusterCondition(ctx context.Context, cluster *cluste
 }
 
 func (c *Controller) updateClusterTaints(ctx context.Context, cluster *clusterapi.ManagedCluster) error {
+	if cluster == nil {
```

Comment:
Could we modify L177 to return non-nil `*clusterapi.ManagedCluster`? If so, we don't need to check whether `cluster` is nil or not?

Reply:
Also please check `if len(cluster.Status.Conditions) > 0`.

Reply:
> I modify L177 firstly, but I considered later that if currentReadyCondition is equal with observedReadyCondition or observedReadyCondition is nil, updateClusterTaints dosn't neet to run, so I add a judgement here.

This will make this piece of codes confusing, especially without any comments. Also this will make it harder to add further changes.

I'd prefer to leave that to `updateClusterTaints`.

Reply:
> len(cluster.Status.Conditions) will never be 0, because it is updated at
https://github.com/clusternet/clusternet/blob/main/pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go#L188-L189

NO!

It may return from here https://github.com/clusternet/clusternet/blob/main/pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go#L176-L178.

Reply:
> > I modify L177 firstly, but I considered later that if currentReadyCondition is equal with observedReadyCondition or observedReadyCondition is nil, updateClusterTaints dosn't neet to run, so I add a judgement here.
> 
> This will make this piece of codes confusing, especially without any comments. Also this will make it harder to add further changes.
> 
> I'd prefer to leave that to `updateClusterTaints`.

ok, move it to L177

Reply:
> > len(cluster.Status.Conditions) will never be 0, because it is updated at
> > https://github.com/clusternet/clusternet/blob/main/pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go#L188-L189
> 
> NO!
> 
> It may return from here https://github.com/clusternet/clusternet/blob/main/pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go#L176-L178.

cluster.Status.Conditions is nil dosen't impact "for range",  please ref: https://thanhtungvo.medium.com/golang-can-iterating-loop-for-nil-slice-ff0727911f54